libevmcore/Assembly.cpp

```diff
@@ -171,61 +173,96 @@ ostream& dev::eth::operator<<(ostream& _out, AssemblyItemsConstRef _i)
 	return _out;
 }
 
-ostream& Assembly::streamRLP(ostream& _out, string const& _prefix) const
+string Assembly::getLocationFromSources(StringMap const& _sourceCodes, SourceLocation const& _location) const
+{
+	if (_location.isEmpty() || _sourceCodes.empty())
+		return "";
+
+	string toReturn = _sourceCodes.find(*_location.sourceName)->second;
+
+	if (_location.end < (int)toReturn.size() - 1)
```

Comment:
I think this should rather check for a valid `_start`, as `substr` throws for an invalid start. Furthermore, you could also check for a valid location (start nonnegative and smaller than end) here.